src/plugins/pendo/components/OptInModal/OptInModal.tsx

```diff
@@ -0,0 +1,159 @@
+import {
+  Button,
+  Divider,
+  List,
+  ListIcon,
+  ListItem,
+  ModalBody,
+  ModalFooter,
+  useColorModeValue,
+} from '@chakra-ui/react'
+import { getConfig } from 'config'
+import { launch } from 'plugins/pendo'
+import { VisitorDataManager } from 'plugins/pendo/visitorData'
+import { useEffect } from 'react'
+import { IoMdCheckmark, IoMdClose } from 'react-icons/io'
+import { useHistory } from 'react-router-dom'
+import { Text } from 'components/Text'
+import { logger } from 'lib/logger'
+
+import { OptInIcon } from './OptInIcon'
+
+const moduleLogger = logger.child({ namespace: ['Plugins', 'Pendo', 'components', 'OptInModal'] })
+
+export const OptInModal: React.FC = () => {
+  const CONSENT_TAG = `pendo_${getConfig().REACT_APP_PENDO_CONSENT_VERSION}`
+
+  const borderColor = useColorModeValue('gray.100', 'gray.750')
+  const history = useHistory()
+
+  const consent = VisitorDataManager.checkConsent(CONSENT_TAG)
+
+  useEffect(() => {
+    if (typeof consent === 'boolean') {
+      moduleLogger.trace({ consent }, 'User has selected their consent')
+      if (consent) launch()
```

Comment:
Is there any other way to handle giving consent outside of the `handleConfirm`? It looks a bit like the possibility to call `launch()` twice when handling confirm and updating consent to true.

Also, not super familiar with the `history` object, but is location state a part of that and could cause refiring of this hook? do we need to check for history at `/` or something?

Reply:
`launch` is a memoized function. No matter how many times you call it, it will only run once.

I don't understand the second part of your question.

Reply:
Was just wondering if the history dependency would cause a re-render or something, but as it is actually changing what component is being rendered it shouldn't matter I don't think